tests/data/random/walkSpec.js

```diff
@@ -0,0 +1,29 @@
+describe('walk', function() {
+
+    beforeEach(function() {
+        // create a random number generator that produces a predictable sequence
+        // 0.2, 0.8, 0.2, 0.8 ...
+        var index = 0;
+        spyOn(d3.random, 'normal').and.returnValue(function() {
+            return index++ % 2 ? 0.2 : 0.8;
+        });
+    });
+
+    it('should output the correct number sequence', function() {
+        var walk = fc.data.random.walk()
+            .steps(5);
+
+        var walkData = walk(10);
+
+        // this test data was generted from the random walk at the following point:
+        // b0ee0a397a6ecc4cf60663d5b51af15de3534ef1
+        // and is assumed correct for the purposes of further refactor.
+        expect(walkData)
+            .toEqual([10, 10.563051249613967, 10.862390962256633, 11.473999242766038, 11.799153741691457]);
```

Comment:
Should the output include the initial price? If so, I'd expect the output to have `steps` + 1 length if `steps` is the number of price increments. E.g. for steps = 1, period = T, Initial price S<sub>0</sub>, the output should be either [S<sub>0</sub>, S<sub>T</sub>] or [S<sub>T</sub>], but not [S<sub>0</sub>]. The current documentation for `steps` is a bit misleading here.